### DISCUSSION TITLE
Property Names

### DISCUSSION THREAD

**Discussion by pwylcrow** [dotnet/aspnet-api-versioning#967] [General]
Property names w/ OData API versioning are being serialized into camelCase when I would like them to be PascalCase.  All attempts at overriding this have failed.  Please advise on how to accomplish this with your library.

**Comment by commonsensesoftware** [comment]
I'm not sure what _"All attempts..."_ included, but OData has its own method for configuring casing. The default is Pascalcase, but most people expect Camelcase as the de facto for JSON, which is also the default for ASP.NET Core. API Versioning aligns to Camelcase by default as part of the _80/20_ rule with the intent you have less to configure.

Reverting to the default settings is as simple as:


```c#
builder.Services.AddControllers().AddOData();
builder.Services.AddApiVersioning()
                .AddOData(
                 options =>
                 {
                     // the default factory method includes .EnableLowerCamelCase()
                     options.ModelBuilder.ModelBuilderFactory = () => new ODataConventionModelBuilder();
                     options.AddRouteComponents();
                 });          
```

The [Versioned OData Model Builder](../wiki/Versioned-ODataModelBuilder) topic provides additional documentation and the builder is exposed via `ODataApiVersioningOptions.ModelBuilder`.

**Reply by pwylcrow** [reply]
Thank you sir
